### DISCUSSION TITLE
About auto convert type

### DISCUSSION THREAD

**Discussion by abersheeran** [pydantic/pydantic#3635] [Question] [answered]
I have the follow definition:
```python
literal: Optional[Union[int, float, str, bool]]
```

If use it to judge `"123"`, then would get `123` (`int`), but I want to get `"123"`, so how should I do.

**Comment by PrettyWood** [comment, marked as answer]
Hi @abersheeran 
You can use [`Config.smart_union`](https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union) that has been added in v1.9 ;)

**Reply by abersheeran** [reply]
Woooooo! Thank you.
